source/common/protobuf/utility.h

```diff
@@ -16,6 +16,20 @@
   ((message).has_##field_name() ? (message).field_name().value()                                   \
                                 : throw MissingFieldException(#field_name, (message)))
 
+// Obtain the milliseconds value of a google.protobuf.Duration field if set. Otherwise, return the
+// default value.
+#define PROTOBUF_GET_MS_OR_DEFAULT(message, field_name, default_value)                             \
+  ((message).has_##field_name()                                                                    \
+       ? Protobuf::util::TimeUtil::DurationToMilliseconds((message).field_name())                  \
+       : (default_value))
+
+// Obtain the milliseconds value of a google.protobuf.Duration field if set. Otherwise, throw a
+// MissingFieldException.
+#define PROTOBUF_GET_MS_REQUIRED(message, field_name)                                              \
+  ((message).has_##field_name()                                                                    \
+       ? Protobuf::util::TimeUtil::DurationToMilliseconds((message).field_name())                  \
+       : throw MissingFieldException(#field_name, (message)))
+
 namespace Envoy {
 
 class MissingFieldException : public EnvoyException {
```

Comment:
Unrelated but I noticed in my coverage glance that MissingFieldException is not covered. Is this used?